applications/nrf_desktop/src/modules/watchdog.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/watchdog.h>
+
+#define MODULE watchdog
+#include "module_state_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_WATCHDOG_LOG_LEVEL);
+
+
+#if defined(CONFIG_WDT_0_NAME)
+#define WATCHDOG_DEV_NAME CONFIG_WDT_0_NAME
+#elif defined(DT_WDT_0_NAME)
+#define WATCHDOG_DEV_NAME DT_WDT_0_NAME
+#else
+#error "Error!!! Watchdog device name not defined - verify configuration."
+#endif
+#define WDT_FEED_WORKER_DELAY_MS 2000
+#define WDT_FEED_WORKER_START_IMMEDIATELY 0
+
+#if defined(CONFIG_DESKTOP_WATCHDOG_TIMEOUT)
+#define WDT_TIMEOUT_MS K_SECONDS(CONFIG_DESKTOP_WATCHDOG_TIMEOUT)
+#else
+/* 16s - required to survive LOG_PANIC print when power down */
+#define WDT_TIMEOUT_MS 16000
```

Comment:
this comes from `CONFIG_DESKTOP_WATCHDOG_TIMEOUT` do else condition is not needed.

Reply:
well, it is not needed because kconfig should always provide a value, I've done it to make the code "bulletproof". I assume that you don't like it, so I wil remove it.